src/Sema.zig

```diff
@@ -12321,6 +12352,25 @@ fn analyzeSwitchRuntimeBlock(
             break :h .none;
         };
 
+        if (single_prong) {
+            @branchHint(.unlikely);
+            assert(operand_has_opv);
+
+            term: {
+                if (case_block.instructions.getLastOrNull()) |last_inst| {
+                    if (sema.isNoReturn(last_inst.toRef())) break :term;
+                }
+                _ = try case_block.addNoOp(.unreach);
+            }
```

Comment:
What's going on here? This should be impossible; the ZIR block, and hence the AIR block it generated, is guaranteed to end with a noreturn terminator.